### DISCUSSION TITLE
Is ffmpeg enabled in mpd for windows ?

### DISCUSSION THREAD

**Discussion by Potatof** [MusicPlayerDaemon/MPD#2143] [Q&A] [answered]
Hi every body,

...Because i get a "No such filter plugin: ffmpeg" message.

Full messages queue  
D:\mpd>mpd -v mpd.conf
config_file: loading file mpd.conf
Oct 31 08:12 : vorbis: Xiph.Org libVorbis 1.3.7
Oct 31 08:12 : opus: libopus 1.4
Oct 31 08:12 : decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Oct 31 08:12 : exception: Failed to access M:PopFleetwood Mac1979 - Tusk (2015 Remaster): Le fichier spécifié est introuvable.
: Le fichier spécifié est introuvable
Oct 31 08:12 : simple_db: reading DB
Oct 31 08:12 : output: Failed to initialize filter chain for 'ffmpeg SuperEQ': No such filter plugin: ffmpeg
Oct 31 08:12 : curl: version 8.5.0
Oct 31 08:12 : curl: with Schannel
Oct 31 08:12 : input: Input plugin 'ffmpeg' is unavailable: No protocol

And how to get the list of enabled plugins ?

Thx a lot

**Comment by MaxKellermann** [comment, marked as answer]
`mpd --version`

**Reply by Potatof** [reply]
Ok,
ffmpeg as input but not as filter.
Too bad that's exactly what I wanted to test :-(
Thx Max

Music Player Daemon 0.23.15 (0.23.15)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple proxy

Storage plugins:
 local nfs curl


Decoders plugins:
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [dsdiff] dff
 [dsf] dsf
 [hybrid_dsd] m4a
 [openmpt] mptm mod s3m xm it 669 amf ams c67 dbm digi dmf dsm dtm far imf ice j2b m15 mdl med mms mt2 mtm nst okt plm psm pt36 ptm sfx sfx2 st26 stk stm stp ult wow gdm mo3 oxm umx xpk ppm mmcmp
 [modplug] 669 amf ams dbm dfm dsm far it med mdl mod mtm mt2 okt s3m stm ult umx xm
 [wildmidi] mid
 [gme] ay gbs gym hes kss nsf nsfe rsn sap spc vgm vgz
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve
 [pcm]

Filters:


Tag plugins:


Output plugins:
 null jack httpd snapcast recorder winmm wasapi

Encoder plugins:
 null vorbis opus lame wave flac

Input plugins:
 file curl ffmpeg nfs

Playlist plugins:
 extm3u m3u pls xspf asx rss flac cue embcue

Protocols:
 http:// https:// nfs://

Other features:
 ipv6 tcp

**Reply by MaxKellermann** [reply]
That means our FFmpeg build doesn't enable the libavfilter component.
This is how we build FFmpeg: https://github.com/MusicPlayerDaemon/MPD/blob/master/python/build/libs.py#L78
If you can figure this out, send a PR.

**Reply by Potatof** [reply]
Okay Max, I'm not sure I'll take this on (compile a personal version for windows) , but you never know and if I do, I'll report back here.

**Comment by WestFlow127** [comment]
Why is it that the MPD [docs](https://mpd.readthedocs.io/en/stable/plugins.html#filter-plugins) say you can use FFmpeg as a filter when it's not actually included as one?

**Reply by WestFlow127** [reply]
Is it included in Mac or linux distros?

**Reply by MaxKellermann** [reply]
MPD is an open source project. You get the source code, and the documentation describes what you can do with, always assuming you have the required compile-time options enabled. MPD has many compile-time options.
Additionally, third-party software such as FFmpeg has many compile-time options. But third-party software is out of MPD's scope. You won't get support for third-party software here.

**Reply by WestFlow127** [reply]
Got it, that all makes sense. I appreciate the clarification and the work that you all do on these projects.

**Comment by leigh123linux** [comment]
@WestFlow127  If you want ffmpeg filter support you will need to build mpd with it enabled in ffmpeg

https://github.com/MusicPlayerDaemon/MPD/blob/master/python/build/libs.py#L78

Linux distro's use external ffmpeg so isn't limited
